### PR TITLE
step4

### PR DESCRIPTION
**Pull request by BogdanPerekhrest34**
checkout controller
cart controller
order_item controller
user/settigns -> privacy and addresses controller
omniauth controller

### CODE REVIEW

File: app/services/users/settings/address_updater.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+      private
+
+      def address
+        @address = Address.where(type: params[:type], user_id: user_id).first_or_initialize
+      end
+
+      def fill_in_address
+        VARIABLES.each do |var|
+          @address.send("#{var}=", params[var]) if params[var]
```
Comment (nico-saj) [inline app/services/users/settings/address_updater.rb:29]: Don't use `send` method here. U can use it only in tests and for debugging. U can use `public_send`

Comment (nico-saj) [reply app/services/users/settings/address_updater.rb:29]: but it will be better to permit params and assign_attributes to address

**Review by nico-saj — CHANGES_REQUESTED** [review]
Add specs

**Comment by BogdanPerekhrest34** [comment]
all specs in step6

**Review by nico-saj — APPROVED** [review]

**Pull request merged**